docs/maintainers/Release_Process.md

```diff
@@ -1,48 +1,64 @@
-Should be done several weeks before planned release date
+### Initial release setup (major releases only)
+Should be done immediately after start of stabilization stage for previous release
+
+- Create project named `Release 1.X`
+- Add all features and bugs that should be fixed as part of this release into this project
+
+### Start of stabilization stage (major releases only)
+Should be done 2-4 weeks before planned release date. All major features should be finished at this point.
 
-- Create [milestone](https://github.com/vcmi/vcmi/milestones) named `Release 1.x`
-- Specify new milestone for all regression bugs and major bugs related to new functionality
 - Create `beta` branch from `develop`
-- Bump vcmi version on `develop` branch
-- Bump version for linux on `develop` branch
+- Bump vcmi version in CMake on `develop` branch
+- Bump version for Linux on `develop` branch
+- Bump version and build ID for Android on `develop` branch
+
+### Initial release setup (patch releases only)
+
+- Bump vcmi version in CMake on `beta` branch
+- Bump version for Linux on `beta` branch
+- Bump version and build ID for Android on `beta` branch
+
+### Release preparation stage
+Should be done 1 week before release. Release date should be decided at this point.
+
+- Make sure to announce codebase freeze deadline (1 day before release) to all developers
+- Create pull request for release preparation tasks targeting `beta`:
+- - Update [changelog](https://github.com/vcmi/vcmi/blob/develop/ChangeLog.md)
+- - Update release date for Linux packaging. See [example](https://github.com/vcmi/vcmi/pull/1258)
+- - Update build ID for Android packaging. See [example](https://github.com/vcmi/vcmi/pull/2090)
+- - Update downloads counter in readme.md. See [example](https://github.com/vcmi/vcmi/pull/2091)
 
 ### Release preparation stage
-Should be done several days before planned release date
-
-- Make sure to announce codebase freeze deadline to all developers
-- Update [release notes](https://github.com/vcmi/vcmi/blob/develop/ChangeLog.md)
-- Update release date for Linux packaging. See [example](https://github.com/vcmi/vcmi/pull/1258)
-- Update release date for Android packaging. See [example](https://github.com/vcmi/vcmi/pull/2090)
-- Create draft release page, specify `1.x.0` as tag for `master` after publishing
-- Prepare pull request with release update for web site https://github.com/vcmi/VCMI.eu
+Should be done 1 day before release. At this point beta branch is in full freeze.
+
+- Merge release preparation PR into `beta`
+- Merge `beta` into `master`. This will trigger CI pipeline that will generate release packages
+- Create draft release page, specify `1.x.y` as tag for `master` after publishing
+- Check that artifacts for all platforms have been build by CI `master` branch
+- Attach build artifacts for all platforms to release page
+- Manually extract Windows installer, remove `$PLUGINSDIR` directory and repackage data as .zip archive
```

Comment:
Windows installer part? Some players have requested "Installer-free" / portable version in form of zip archive. And according to download stats it does have popularity (accounts for 25% of Windows downloads)

I don't know how to make Windows CI generate multiple artifacts from single run. If we can make it automatically during CI instead of manual actions - that would be great.

Reply:
Sorry, I mean the removal of plugins dir.

As for multiple artifacts, some cmake adjustments would be needed, as cpack generates only the installer. To get zip, there's a built-in cpack generator, just need to figure out how to pass generator type when cpack is called.

Reply:
Ah, `$PLUGINSDIR` is internal directory that NSIS installer places into archive (but does not installs). It contains resources used by installer itself.